inspector: Fix failing test

eugeneo opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
This change only touches inspector test.

##### Description of change
Test was updated to wait till the inspector processes socket closure.

CC: @ofrobots

addaleax commented on test/cctest/test_inspector_socket.cc:
```diff
@@ -298,7 +298,7 @@ static void really_close(uv_handle_t* handle) {
 static void manual_inspector_socket_cleanup() {
   EXPECT_EQ(0, uv_is_active(
-                   reinterpret_cast<uv_handle_t*>(&inspector.client)));
+            reinterpret_cast<uv_handle_t*>(&inspector.client)));
```
This looked okay to me? `reinterpret_cast<…>()` is an argument to `uv_is_active`, not `EXPECT_EQ`

eugeneo replied:
Right. I reverted the change.

ofrobots commented:
nit: since this is fixing a test, the commit message should probably use the 'test: ' prefix. Otherwise LGTM. 

CI: https://ci.nodejs.org/job/node-test-pull-request/3582/. I have verified locally that this fixes #8006.

/cc @bnoordhuis as well.

eugeneo commented:
Thank you for the review. I updated the commit message.

bnoordhuis commented:
LGTM but can you update the commit log status line to e.g. `test: fix failing inspector cctest`?

I admit I don't understand why the test passes for me locally with and without this change.

eugeneo commented:
@bnoordhuis I updated the message.

This test failure was intermittent - it looks like sockets closing on Mac is somehow more "async" then on other systems (or simply slower) so this test failed because the cleanup code saw inspector socket as still open. If I added a delay (e.g. printed a message to stderr) the test succeeded.

ofrobots commented:
Landed as 49e473a.
